### PR TITLE
PR: Radial derivatives fix in Dommaschk potentials

### PR DESCRIPTION
**Pull request by migueljmp**
TO DO: Fix tests

### CODE REVIEW

**Comment by migueljmp** [comment]
Hello, I was working on a project with @rogeriojorge  using Dommaschk potentials, which led me to verify my derivatives of the potentials in the paper with the example here on simsopt. As it turns out, there was a term missing from the current version, which is only non-zero when the poloidal periodicity l is double or more than double the toroidal periodicity m. Any time there was a derivative by R, there was a term that appeared because of the log(R) term, and in the double derivatives, that occurred twice. I had a strange issue where I couldn't simplify the terms to call the alpha functions fewer times, the double radial derivatives were diverging, so in its current state there's some inefficiency there, I'm probably missing a parenthesis somehow but I left the diverging version as a comment.
Currently, there is no test for the cases that I'm mentioning within the test_magneticfields.py file, I just checked manually against a Mathematica notebook.

**Review by rogeriojorge — COMMENTED** [review]
Thanks for this fix @migueljmp! Some comments attached

**Review by rogeriojorge — APPROVED** [review]

**Review by mbkumar — APPROVED** [review]

**Comment by mbkumar** [comment]
Can we get unit tests as Matt mentioned?

**Comment by migueljmp** [comment]
@landreman @mbkumar As requested, I've added 3 tests that I think have some edge cases, like l=0 l=19, and also just a test for the field configuration in the Dommaschk paper.

**Review by landreman — DISMISSED** [review]

File: tests/field/test_magneticfields.py [hunk 1]
```diff
@@ -474,6 +474,52 @@ def test_Dommaschk(self):
...
+        coeffs = [[1.4, 1.4], [19.25, 0], [5.10e10, 5.10e10]]
+        Bfield = Dommaschk(mn=mn, coeffs=coeffs)
+        point = np.asarray([[0.71879008, 0.76265643, 0.0745]])
+        Bfield.set_points(point)
+        gradB = np.array(Bfield.dB_by_dX())
+        transpGradB = np.array([dBdx.T for dBdx in gradB])
+        B = Bfield.B()
+        print(point)
+        print(B)
+        print(gradB)
```
Comment (rogeriojorge) [inline tests/field/test_magneticfields.py:489]: Could you just remove these prints? I can then approve it and merge it if everyone is ok with it

**Comment by migueljmp** [comment]
The commit "Removed prints" is the one I wanted to commit last, the other one was an accident and seems to have undone the merging from master, I am sorry

**Comment by rogeriojorge** [comment]
Looks good to merge on my side. Given the one "tests" file that prevents an automatic merge, perhaps @mbkumar or @landreman could you do the merge?

**Comment by mbkumar** [comment]
Sure

Bharat Medasani

Engineer
Princeton Plasma Physics Lab (PPPL)

**Review by mbkumar — APPROVED** [review]
